crates/flipperzero/src/furi/stream_buffer.rs

```diff
@@ -0,0 +1,439 @@
+use core::{ffi::c_void, fmt::Debug, num::NonZeroUsize, ops::Deref, ptr::NonNull};
+
+use alloc::sync::Arc;
+
+use flipperzero_sys::{self as sys, furi::Status};
+use ufmt::{derive::uDebug, uDebug};
+
+/// Basic Result which only indicates whether something succeeded or not.
+type EmptyResult = Result<(), ()>;
+
+// https://github.com/flipperdevices/flipperzero-firmware/blob/b723d463afccf628712475e11a3d4579f0331f5c/furi/core/base.h#L13
+const FURI_WAIT_FOREVER: u32 = 0xFFFFFFFF;
```

Comment:
It would probably be good to have this in `furi::time`. There is a `MAX_DURATION_TICKS` (`0x7FFFFFFFF`) though that has a slightly different purpose.

Reply:
I added `WAIT_FOREVER` as a new constant.